### PR TITLE
[ws] fix missing type definitions in ClientOptions

### PR DESCRIPTION
**Pull request by seompark**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketaddress-protocols-options>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ws/index.d.ts [hunk 1]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    once(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
+    once(event: 'close', listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
+    once(event: 'error', listener: (this: WebSocket, err: Error) => void): this;
+    once(event: 'upgrade', listener: (this: WebSocket, request: IncomingMessage) => void): this;
+    once(event: 'message', listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
+    once(event: 'open', listener: (this: WebSocket) => void): this;
+    once(event: 'ping' | 'pong', listener: (this: WebSocket, data: Buffer) => void): this;
     once(
-        event: "unexpected-response",
+        event: 'unexpected-response',
```
Comment (vansergen) [inline types/ws/index.d.ts:137]:
```suggestion
        event: "unexpected-response",
```

File: types/ws/index.d.ts [hunk 2]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    off(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
-    off(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
-    off(event: "open", listener: (this: WebSocket) => void): this;
-    off(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
+    off(event: 'close', listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
+    off(event: 'error', listener: (this: WebSocket, err: Error) => void): this;
+    off(event: 'upgrade', listener: (this: WebSocket, request: IncomingMessage) => void): this;
+    off(event: 'message', listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
+    off(event: 'open', listener: (this: WebSocket) => void): this;
+    off(event: 'ping' | 'pong', listener: (this: WebSocket, data: Buffer) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:147]:
```suggestion
    off(event: "close", listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
    off(event: "error", listener: (this: WebSocket, err: Error) => void): this;
    off(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
    off(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
    off(event: "open", listener: (this: WebSocket) => void): this;
    off(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
```

File: types/ws/index.d.ts [hunk 3]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
     off(
-        event: "unexpected-response",
+        event: 'unexpected-response',
```
Comment (vansergen) [inline types/ws/index.d.ts:149]:
```suggestion
        event: "unexpected-response",
```

File: types/ws/index.d.ts [hunk 4]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    addListener(event: "upgrade", listener: (request: IncomingMessage) => void): this;
-    addListener(event: "message", listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
-    addListener(event: "open", listener: () => void): this;
-    addListener(event: "ping" | "pong", listener: (data: Buffer) => void): this;
+    addListener(event: 'close', listener: (code: number, reason: Buffer) => void): this;
+    addListener(event: 'error', listener: (err: Error) => void): this;
+    addListener(event: 'upgrade', listener: (request: IncomingMessage) => void): this;
+    addListener(event: 'message', listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
+    addListener(event: 'open', listener: () => void): this;
+    addListener(event: 'ping' | 'pong', listener: (data: Buffer) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:159]:
```suggestion
    addListener(event: "close", listener: (code: number, reason: Buffer) => void): this;
    addListener(event: "error", listener: (err: Error) => void): this;
    addListener(event: "upgrade", listener: (request: IncomingMessage) => void): this;
    addListener(event: "message", listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
    addListener(event: "open", listener: () => void): this;
    addListener(event: "ping" | "pong", listener: (data: Buffer) => void): this;
```

File: types/ws/index.d.ts [hunk 5]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
     addListener(
-        event: "unexpected-response",
+        event: 'unexpected-response',
```
Comment (vansergen) [inline types/ws/index.d.ts:161]:
```suggestion
        event: "unexpected-response",
```

File: types/ws/index.d.ts [hunk 6]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    removeListener(event: "upgrade", listener: (request: IncomingMessage) => void): this;
-    removeListener(event: "message", listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
-    removeListener(event: "open", listener: () => void): this;
-    removeListener(event: "ping" | "pong", listener: (data: Buffer) => void): this;
+    removeListener(event: 'close', listener: (code: number, reason: Buffer) => void): this;
+    removeListener(event: 'error', listener: (err: Error) => void): this;
+    removeListener(event: 'upgrade', listener: (request: IncomingMessage) => void): this;
+    removeListener(event: 'message', listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
+    removeListener(event: 'open', listener: () => void): this;
+    removeListener(event: 'ping' | 'pong', listener: (data: Buffer) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:171]:
```suggestion
    removeListener(event: "close", listener: (code: number, reason: Buffer) => void): this;
    removeListener(event: "error", listener: (err: Error) => void): this;
    removeListener(event: "upgrade", listener: (request: IncomingMessage) => void): this;
    removeListener(event: "message", listener: (data: WebSocket.RawData, isBinary: boolean) => void): this;
    removeListener(event: "open", listener: () => void): this;
    removeListener(event: "ping" | "pong", listener: (data: Buffer) => void): this;
```

File: types/ws/index.d.ts [hunk 7]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
     removeListener(
-        event: "unexpected-response",
+        event: 'unexpected-response',
```
Comment (vansergen) [inline types/ws/index.d.ts:173]:
```suggestion
        event: "unexpected-response",
```

File: types/ws/index.d.ts [hunk 8]
```diff
@@ -210,6 +215,7 @@ declare namespace WebSocket {
     interface ClientOptions extends SecureContextOptions {
         protocol?: string | undefined;
         followRedirects?: boolean | undefined;
+        generateMask?: (mask: Buffer) => void | undefined;
```
Comment (vansergen) [inline types/ws/index.d.ts:218]:
```suggestion
        generateMask?: (mask: Buffer) => void;
```

File: types/ws/index.d.ts [hunk 9]
```diff
@@ -223,24 +229,27 @@ declare namespace WebSocket {
...
+                  finishFlush?: number | undefined;
+                  chunkSize?: number | undefined;
+                  windowBits?: number | undefined;
+                  level?: number | undefined;
+                  memLevel?: number | undefined;
+                  strategy?: number | undefined;
+                  dictionary?: Buffer | Buffer[] | DataView | undefined;
+                  info?: boolean | undefined;
+              }
+            | undefined;
```
Comment (vansergen) [inline types/ws/index.d.ts:252]:
```suggestion
        zlibDeflateOptions?: {
            flush?: number | undefined;
            finishFlush?: number | undefined;
            chunkSize?: number | undefined;
            windowBits?: number | undefined;
            level?: number | undefined;
            memLevel?: number | undefined;
            strategy?: number | undefined;
            dictionary?: Buffer | Buffer[] | DataView | undefined;
            info?: boolean | undefined;
        } | undefined;
```

File: types/ws/index.d.ts [hunk 10]
```diff
@@ -316,34 +325,34 @@ declare namespace WebSocket {
...
 
         // Events
-        on(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
-        on(event: "error", cb: (this: Server, error: Error) => void): this;
-        on(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
-        on(event: "close" | "listening", cb: (this: Server) => void): this;
+        on(event: 'connection', cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
+        on(event: 'error', cb: (this: Server, error: Error) => void): this;
+        on(event: 'headers', cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
+        on(event: 'close' | 'listening', cb: (this: Server) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:331]:
```suggestion
        on(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
        on(event: "error", cb: (this: Server, error: Error) => void): this;
        on(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
        on(event: "close" | "listening", cb: (this: Server) => void): this;
```

File: types/ws/index.d.ts [hunk 11]
```diff
@@ -316,34 +325,34 @@ declare namespace WebSocket {
...
         on(event: string | symbol, listener: (this: Server, ...args: any[]) => void): this;
 
-        once(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
-        once(event: "error", cb: (this: Server, error: Error) => void): this;
-        once(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
-        once(event: "close" | "listening", cb: (this: Server) => void): this;
+        once(event: 'connection', cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
+        once(event: 'error', cb: (this: Server, error: Error) => void): this;
+        once(event: 'headers', cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
+        once(event: 'close' | 'listening', cb: (this: Server) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:337]:
```suggestion
        once(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
        once(event: "error", cb: (this: Server, error: Error) => void): this;
        once(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
        once(event: "close" | "listening", cb: (this: Server) => void): this;
```

File: types/ws/index.d.ts [hunk 12]
```diff
@@ -316,34 +325,34 @@ declare namespace WebSocket {
...
         once(event: string | symbol, listener: (...args: any[]) => void): this;
 
-        off(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
-        off(event: "error", cb: (this: Server, error: Error) => void): this;
-        off(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
-        off(event: "close" | "listening", cb: (this: Server) => void): this;
+        off(event: 'connection', cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
+        off(event: 'error', cb: (this: Server, error: Error) => void): this;
+        off(event: 'headers', cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
+        off(event: 'close' | 'listening', cb: (this: Server) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:343]:
```suggestion
        off(event: "connection", cb: (this: Server, socket: WebSocket, request: IncomingMessage) => void): this;
        off(event: "error", cb: (this: Server, error: Error) => void): this;
        off(event: "headers", cb: (this: Server, headers: string[], request: IncomingMessage) => void): this;
        off(event: "close" | "listening", cb: (this: Server) => void): this;
```

File: types/ws/index.d.ts [hunk 13]
```diff
@@ -316,34 +325,34 @@ declare namespace WebSocket {
...
         off(event: string | symbol, listener: (this: Server, ...args: any[]) => void): this;
 
-        addListener(event: "connection", cb: (client: WebSocket, request: IncomingMessage) => void): this;
-        addListener(event: "error", cb: (err: Error) => void): this;
-        addListener(event: "headers", cb: (headers: string[], request: IncomingMessage) => void): this;
-        addListener(event: "close" | "listening", cb: () => void): this;
+        addListener(event: 'connection', cb: (client: WebSocket, request: IncomingMessage) => void): this;
+        addListener(event: 'error', cb: (err: Error) => void): this;
+        addListener(event: 'headers', cb: (headers: string[], request: IncomingMessage) => void): this;
+        addListener(event: 'close' | 'listening', cb: () => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:349]:
```suggestion
        addListener(event: "connection", cb: (client: WebSocket, request: IncomingMessage) => void): this;
        addListener(event: "error", cb: (err: Error) => void): this;
        addListener(event: "headers", cb: (headers: string[], request: IncomingMessage) => void): this;
        addListener(event: "close" | "listening", cb: () => void): this;
```

File: types/ws/index.d.ts [hunk 14]
```diff
@@ -316,34 +325,34 @@ declare namespace WebSocket {
...
         addListener(event: string | symbol, listener: (...args: any[]) => void): this;
 
-        removeListener(event: "connection", cb: (client: WebSocket) => void): this;
-        removeListener(event: "error", cb: (err: Error) => void): this;
-        removeListener(event: "headers", cb: (headers: string[], request: IncomingMessage) => void): this;
-        removeListener(event: "close" | "listening", cb: () => void): this;
+        removeListener(event: 'connection', cb: (client: WebSocket) => void): this;
+        removeListener(event: 'error', cb: (err: Error) => void): this;
+        removeListener(event: 'headers', cb: (headers: string[], request: IncomingMessage) => void): this;
+        removeListener(event: 'close' | 'listening', cb: () => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:355]:
```suggestion
        removeListener(event: "connection", cb: (client: WebSocket) => void): this;
        removeListener(event: "error", cb: (err: Error) => void): this;
        removeListener(event: "headers", cb: (headers: string[], request: IncomingMessage) => void): this;
        removeListener(event: "close" | "listening", cb: () => void): this;
```

File: types/ws/index.d.ts [hunk 15]
```diff
@@ -12,20 +12,20 @@
 
 /// <reference types="node" />
 
-import { EventEmitter } from "events";
+import { EventEmitter } from 'events';
```
Comment (vansergen) [inline types/ws/index.d.ts:15]:
```suggestion
import { EventEmitter } from "events";
```

File: types/ws/index.d.ts [hunk 16]
```diff
@@ -12,20 +12,20 @@
...
-import { Duplex, DuplexOptions } from "stream";
-import { SecureContextOptions } from "tls";
-import { URL } from "url";
-import { ZlibOptions } from "zlib";
+} from 'http';
+import { Server as HTTPSServer } from 'https';
+import { Duplex, DuplexOptions } from 'stream';
+import { SecureContextOptions } from 'tls';
+import { URL } from 'url';
+import { ZlibOptions } from 'zlib';
```
Comment (vansergen) [inline types/ws/index.d.ts:28]:
```suggestion
} from "http";
import { Server as HTTPSServer } from "https";
import { Duplex, DuplexOptions } from "stream";
import { SecureContextOptions } from "tls";
import { URL } from "url";
import { ZlibOptions } from "zlib";
```

File: types/ws/index.d.ts [hunk 17]
```diff
@@ -38,7 +38,7 @@ declare class WebSocket extends EventEmitter {
     /** The connection is closed. */
     static readonly CLOSED: 3;
 
-    binaryType: "nodebuffer" | "arraybuffer" | "fragments";
+    binaryType: 'nodebuffer' | 'arraybuffer' | 'fragments';
```
Comment (vansergen) [inline types/ws/index.d.ts:41]:
```suggestion
    binaryType: "nodebuffer" | "arraybuffer" | "fragments";
```

File: types/ws/index.d.ts [hunk 18]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
     send(data: any, cb?: (err?: Error) => void): void;
     send(
         data: any,
-        options: { mask?: boolean | undefined; binary?: boolean | undefined; compress?: boolean | undefined; fin?: boolean | undefined },
+        options: {
+            mask?: boolean | undefined;
+            binary?: boolean | undefined;
+            compress?: boolean | undefined;
+            fin?: boolean | undefined;
+        },
```
Comment (vansergen) [inline types/ws/index.d.ts:85]:
```suggestion
        options: { mask?: boolean | undefined; binary?: boolean | undefined; compress?: boolean | undefined; fin?: boolean | undefined },
```

File: types/ws/index.d.ts [hunk 19]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
         cb?: (err?: Error) => void,
     ): void;
     terminate(): void;
 
     // HTML5 WebSocket events
     addEventListener(
-        method: "message",
+        method: 'message',
```
Comment (vansergen) [inline types/ws/index.d.ts:92]:
```suggestion
        method: "message",
```

File: types/ws/index.d.ts [hunk 20]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
         cb: (event: WebSocket.MessageEvent) => void,
         options?: WebSocket.EventListenerOptions,
     ): void;
     addEventListener(
-        method: "close",
+        method: 'close',
```
Comment (vansergen) [inline types/ws/index.d.ts:97]:
```suggestion
        method: "close",
```

File: types/ws/index.d.ts [hunk 21]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
         cb: (event: WebSocket.CloseEvent) => void,
         options?: WebSocket.EventListenerOptions,
     ): void;
     addEventListener(
-        method: "error",
+        method: 'error',
```
Comment (vansergen) [inline types/ws/index.d.ts:102]:
```suggestion
        method: "error",
```

File: types/ws/index.d.ts [hunk 22]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
         cb: (event: WebSocket.ErrorEvent) => void,
         options?: WebSocket.EventListenerOptions,
     ): void;
     addEventListener(
-        method: "open",
+        method: 'open',
```
Comment (vansergen) [inline types/ws/index.d.ts:107]:
```suggestion
        method: "open",
```

File: types/ws/index.d.ts [hunk 23]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
     ): void;
 
-    removeEventListener(method: "message", cb: (event: WebSocket.MessageEvent) => void): void;
-    removeEventListener(method: "close", cb: (event: WebSocket.CloseEvent) => void): void;
-    removeEventListener(method: "error", cb: (event: WebSocket.ErrorEvent) => void): void;
-    removeEventListener(method: "open", cb: (event: WebSocket.Event) => void): void;
+    removeEventListener(method: 'message', cb: (event: WebSocket.MessageEvent) => void): void;
+    removeEventListener(method: 'close', cb: (event: WebSocket.CloseEvent) => void): void;
+    removeEventListener(method: 'error', cb: (event: WebSocket.ErrorEvent) => void): void;
+    removeEventListener(method: 'open', cb: (event: WebSocket.Event) => void): void;
```
Comment (vansergen) [inline types/ws/index.d.ts:115]:
```suggestion
    removeEventListener(method: "message", cb: (event: WebSocket.MessageEvent) => void): void;
    removeEventListener(method: "close", cb: (event: WebSocket.CloseEvent) => void): void;
    removeEventListener(method: "error", cb: (event: WebSocket.ErrorEvent) => void): void;
    removeEventListener(method: "open", cb: (event: WebSocket.Event) => void): void;
```

File: types/ws/index.d.ts [hunk 24]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    on(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
-    on(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
-    on(event: "open", listener: (this: WebSocket) => void): this;
-    on(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
+    on(event: 'close', listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
+    on(event: 'error', listener: (this: WebSocket, err: Error) => void): this;
+    on(event: 'upgrade', listener: (this: WebSocket, request: IncomingMessage) => void): this;
+    on(event: 'message', listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
+    on(event: 'open', listener: (this: WebSocket) => void): this;
+    on(event: 'ping' | 'pong', listener: (this: WebSocket, data: Buffer) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:123]:
```suggestion
    on(event: "close", listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
    on(event: "error", listener: (this: WebSocket, err: Error) => void): this;
    on(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
    on(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
    on(event: "open", listener: (this: WebSocket) => void): this;
    on(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
```

File: types/ws/index.d.ts [hunk 25]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
     on(
-        event: "unexpected-response",
+        event: 'unexpected-response',
```
Comment (vansergen) [inline types/ws/index.d.ts:125]:
```suggestion
        event: "unexpected-response",
```

File: types/ws/index.d.ts [hunk 26]
```diff
@@ -77,95 +77,100 @@ declare class WebSocket extends EventEmitter {
...
-    once(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
-    once(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
-    once(event: "open", listener: (this: WebSocket) => void): this;
-    once(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
+    once(event: 'close', listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
+    once(event: 'error', listener: (this: WebSocket, err: Error) => void): this;
+    once(event: 'upgrade', listener: (this: WebSocket, request: IncomingMessage) => void): this;
+    once(event: 'message', listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
+    once(event: 'open', listener: (this: WebSocket) => void): this;
+    once(event: 'ping' | 'pong', listener: (this: WebSocket, data: Buffer) => void): this;
```
Comment (vansergen) [inline types/ws/index.d.ts:135]:
```suggestion
    once(event: "close", listener: (this: WebSocket, code: number, reason: Buffer) => void): this;
    once(event: "error", listener: (this: WebSocket, err: Error) => void): this;
    once(event: "upgrade", listener: (this: WebSocket, request: IncomingMessage) => void): this;
    once(event: "message", listener: (this: WebSocket, data: WebSocket.RawData, isBinary: boolean) => void): this;
    once(event: "open", listener: (this: WebSocket) => void): this;
    once(event: "ping" | "pong", listener: (this: WebSocket, data: Buffer) => void): this;
```

**Review by vansergen — CHANGES_REQUESTED** [review]
@smn-is Could you add some tests for those options?

**Review by k-yle — CHANGES_REQUESTED** [review]
all these formatting changes are going to cause conflicts with the other two open PRs. If you want to get this merged faster, please remove the formatting/style changes

**Review by k-yle — APPROVED** [review]
thanks!

**Review by vansergen — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

File: types/ws/index.d.ts [hunk 27]
```diff
@@ -210,6 +210,7 @@ declare namespace WebSocket {
     interface ClientOptions extends SecureContextOptions {
         protocol?: string | undefined;
         followRedirects?: boolean | undefined;
+        generateMask?: (mask: Buffer) => void;
```
Comment (jakebailey) [inline types/ws/index.d.ts:213]: Should this be:

```suggestion
        generateMask?(mask: Buffer): void;
```

To match the other function in this interfaces below / other interfaces?

**Review by jakebailey — APPROVED** [review]

**Comment by smn-is** [comment]
Ready to merge

**Pull request merged**